### ISSUE TITLE
Annotations apply to lines that they shouldn't

### ISSUE THREAD

**Issue by sadlerjw-bridgit** [krzysztofzablocki/Sourcery#1025]
From the documentation:
> - Annotations at the end of a line will be applied to all declarations on the same line
> - Using `/*` and `*/` for annotation comment you can put annotations on the same line preceding your declaration. This is useful for annotating methods parameters and enum case associated values. All such annotations should be placed in one comment block.

However it appears that these annotations apply both to the line on which they appear as well as line after.

### Environment
```bash
$ swift --version
swift-driver version: 1.26.21 Apple Swift version 5.5.2 (swiftlang-1300.0.47.5 clang-1300.0.29.30)
Target: x86_64-apple-macosx11.0
```

Produced the following example using Sourcery Pro 1.3.4. (But also reproduced the same problem in similar ways on Sourcery 1.0.3 and Sourcery 1.6.1.)

### Example source
```swift
class Foo {
    func bar(paramA: String,
        /* sourcery: testAnnotation="PARAM B ONLY"*/ paramB: String, 
        paramC: String,  // sourcery: anotherAnnotation = "PARAM C ONLY"
        paramD: String
    ) {}
}
```

### Expected result
- only the "paramB" `MethodParameter` should be annotated with `testAnnotation` with the value "PARAM B ONLY
- only the "paramC" `MethodParameter` should be annotated with `anotherAnnotation` with the value "PARAM C ONLY

### Actual result
- `testAnnotation` with the value "PARAM B ONLY" applies to both `paramB` and `paramC`
- `anotherAnnotation` with the value "PARAM C ONLY" applies to both `paramC` and `paramD`

### Workaround
Place Sourcery annotations on their own line, so that they apply only to the line below.

### Example template
```stencil
{% for type in types.all %}
    {{ type.name }}
    {% for method in type.methods %}
        {{method.callName}}
        {% for param in method.parameters %}
            {{ param.name }} annotated with: {{ param.annotations }}
        {% endfor %}
    {% endfor %}
{% endfor %}
```

### Example output
```
Foo
        bar
            paramA annotated with: {
}
            paramB annotated with: {
    testAnnotation = "PARAM B ONLY";
}
            paramC annotated with: {
    anotherAnnotation = "PARAM C ONLY";
    testAnnotation = "PARAM B ONLY";
}
            paramD annotated with: {
    anotherAnnotation = "PARAM C ONLY";
}
```

**Comment by timwredwards** [comment]
+1

**Comment by art-divin** [comment]
👋🏻 Hello,

thank you for your minimal reproducible, @sadlerjw-bridgit 👍🏻 

It really helped. Though, it took me 1 week to debug the issue and 2 days to correctly fix it.
One of the toughest bugs (actually, missing feature) to resolve in my experience working with Sourcery 🎆 

This issue is resolved by #1296

Thanks again, cheers 🍉
